### ISSUE TITLE
FileSystem tests failing on Linux if local timezone != UTC

### ISSUE THREAD

**Issue by nguerrera** [dotnet/corefx#2603]
```
System.IO.FileSystem.Tests.File_GetSetTimes.SettingUpdatesProperties [FAIL]
      Assert.All() Failure: 2 out of 4 items in the collection did not pass.
      [2]: Xunit.Sdk.EqualException: Assert.Equal() Failure
           Expected: 2014-12-01T12:00:00.0000000-08:00
           Actual:   2014-12-01T20:00:00.0000000
```

**Comment by stephentoub** [comment]
cc: @ianhays

**Comment by ianhays** [comment]
Hmm, these tests worked a few days ago. It looks like there was a change in coreclr within the last week that caused the failures - perhaps some update to how localtime is read?

After updating my coreclr build on Linux, I also get failures for ```File_GetSetTimes.CreationSetsAllTimes```, ```Directory_GetSetTimes.CreationSetsAllTimes```, and ```Directory_GetSetTimes.SettingUpdatesProperties```, all occurring with the non-utc setters and getters.

I tried refactoring the tests with different methods of UTC conversion and declaration with different DateTimeKinds, but they don't seem to be working on Linux. I'll investigate it more and try to discern the issue.

**Comment by nguerrera** [comment]
cc @eerhardt @ellismg

**Comment by rqadri133** [comment]
How to unsubscribe from this corefx

On Mon, Aug 3, 2015 at 9:26 AM, Stephen Toub <notifications@github.com>
wrote:

> cc: @ianhays <https://github.com/ianhays>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dotnet/corefx/issues/2603#issuecomment-127232835>.
>



-- 






*Rafi QadriSoftware Engineer240-330-7827Email Security Code=00101*

**Comment by stephentoub** [comment]
> How to unsubscribe from this corefx

@rqadri133, "unwatch" the repo.  There should be a button to do that at the top of the page when you're signed in to GitHub.

**Comment by ianhays** [comment]
Running on Ubuntu 14.04 with local time set to UTC-5, I get the following results to this messy test (in comments):
```
FileInfo testFile = new FileInfo(GetTestFilePath());
testFile.Create().Dispose();

DateTime dt = new DateTime(2014, 12, 1, 12, 0, 0, DateTimeKind.Local);
Console.WriteLine("regular: " + dt.ToString());                              // 12/01/2014 12:00:00
Console.WriteLine("local: " + dt.ToLocalTime().ToString());                  // 12/01/2014 12:00:00
Console.WriteLine("utc: " + dt.ToUniversalTime().ToString());                // 12/01/2014 17:00:00

File.SetLastAccessTime(testFile.FullName, dt);
var res = File.GetLastAccessTime(testFile.FullName);
Console.WriteLine("result: " + res.ToString());                              // 12/01/2014 17:00:00
Console.WriteLine("result.toLocal(): " + res.ToLocalTime().ToString());      // 12/01/2014 12:00:00
Console.WriteLine("result.ToUTC(): " + res.ToUniversalTime().ToString());    // 12/01/2014 22:00:00
Console.WriteLine();

Assert.Equal(dt.ToUniversalTime(), res.ToUniversalTime());                   // fails
```
It would appear that the DateTime being returned by the File and Directory getters (GetLastAccessTime and GetLastWriteTime) is <i>neither</i> UTC nor local. It is supposed to be local and was when I pushed these tests, so we must have made a regression somewhere.

**Comment by eerhardt** [comment]
@ianhays, I made a change last week, https://github.com/dotnet/coreclr/commit/a4f1f97cb58b34d0af3015c5bff16539fc637727 that fixed "Local" time zones on Linux.  Before my change, anytime someone asked for the Local time zone, UTC was always being returned.  After my change, TimeZoneInfo and DateTime.Now started respecting the local time zone the Linux machine was set to.
Ping me offline if you want me to look into this issue.

**Comment by ianhays** [comment]
Thanks for the help @eerhardt, the issue ended up being with the getters not converting their values to  LocalTime like they were supposed to. I checked the values written by the setters and they were fine, so the issue was on my side and just never came up since local times didn't work on Unix before.

Moved the issue to the more specific #2610 now that we now what the problem is. I've got a PR with a fix nearly ready.
